modules/dnn/src/layers/eltwise_layer.cpp

```diff
@@ -374,6 +383,89 @@ class EltwiseLayerImpl CV_FINAL : public EltwiseLayer
                             coeffs, op, activ.get(), nstripes);
     }
 
+#ifdef HAVE_CUDA
+    void forwardCUDA(
+        std::vector<cv::Ptr<BackendWrapper>>& inputs,
+        std::vector<cv::Ptr<BackendWrapper>>& outputs,
+        csl::Workspace& workspace
+    ) override
+    {
+        CV_Assert(outputs.size() == 1);
+
+        auto output_wrapper = outputs[0].dynamicCast<CUDABackendWrapperFP32>();
+        auto output = output_wrapper->getSpan();
+
+        /* optimized path for common case */
+        if (inputs.size() == 2)
+        {
+            auto input_wrapper_x = inputs[0].dynamicCast<CUDABackendWrapperFP32>();
+            auto input_x = input_wrapper_x->getView();
+
+            auto input_wrapper_y = inputs[1].dynamicCast<CUDABackendWrapperFP32>();
+            auto input_y = input_wrapper_y->getView();
+
+            switch(op)
+            {
+            case MAX: csl::kernels::eltwise_max_2<float>(stream, output, input_x, input_y); break;
+            case PROD: csl::kernels::eltwise_prod_2<float>(stream, output, input_x, input_y); break;
+            case SUM:
+                if (coeffs.empty() || (coeffs[0] == 1 && coeffs[1] == 1))
+                    csl::kernels::eltwise_sum_2<float>(stream, output, input_x, input_y);
+                else
+                    csl::kernels::eltwise_sum_coeff_2<float>(stream, output, coeffs[0], input_x, coeffs[1], input_y);
+                break;
+            }
+        }
+        else
+        {
+            auto input_wrapper_0 = inputs[0].dynamicCast<CUDABackendWrapperFP32>();
+            auto input_0 = input_wrapper_0->getView();
+
+            /* we first make a copy and then apply EltwiseOp cumulatively */
+            csl::tensor_ops::copy(stream, output, input_0);
```

Comment:
I have added an optimized code path for the common case where there are just two inputs.

In the general case, the first input is copied to the output. The operation is applied pairwise cumulatively with `output` and `input[i]` as the operands. This may not be the most efficient way.

An optimized solution could reduce the number of memory accesses by a factor of two.

It's possible to parallely peform the operation on all inputs. A `pointerTensor` (can use workspace memory which is otherwise unused in this layer) would hold the addresses of all the input tensors. A kernel would perform the operation on all inputs simultaneously.